game-core/src/main/java/org/triplea/game/server/HeadlessServerSetup.java

```diff
@@ -32,54 +35,75 @@
-    createLobbyWatcher();
+
+    try {
+      createLobbyWatcher();
+    } catch (final FeignException e) {
+      throw new CouldNotConnectToLobby(e);
+    }
   }
 
   private void createLobbyWatcher() {
+    final URI lobbyUri = lobbyUriFromSystemProps();
+
+    final GameHostingResponse gameHostingResponse =
+        GameHostingClient.newClient(lobbyUri).sendGameHostingRequest();
+
+    final GameListingClient gameListingClient =
+        GameListingClient.newClient(lobbyUri, ApiKey.of(gameHostingResponse.getApiKey()));
+
     final InGameLobbyWatcher watcher =
         InGameLobbyWatcher.newInGameLobbyWatcher(
-                model.getMessenger(), lobbyWatcher.getInGameLobbyWatcher())
+                model.getMessenger(),
+                gameHostingResponse,
+                gameListingClient,
+                log::warning,
+                log::info,
+                lobbyWatcher.getInGameLobbyWatcher())
             .orElseThrow(CouldNotConnectToLobby::new);
 
     lobbyWatcher.setInGameLobbyWatcher(watcher);
     lobbyWatcher.setGameSelectorModel(gameSelectorModel);
 
+    final HttpLobbyClient httpLobbyClient =
+        HttpLobbyClient.newClient(lobbyUri, ApiKey.of(gameHostingResponse.getApiKey()));
     LocalServerAvailabilityCheck.builder()
-        .connectivityCheckClient(
-            HttpLobbyClient.newClient(
-                    URI.create(
-                        HttpLobbyClient.PROTOCOL
-                            + System.getProperty(LOBBY_HOST)
-                            + ":"
-                            + System.getProperty(LOBBY_HTTPS_PORT)),
-                    watcher.getLobbyMessenger().getApiKey())
-                .getConnectivityCheckClient())
+        .connectivityCheckClient(httpLobbyClient.getConnectivityCheckClient())
         .localPort(model.getMessenger().getLocalNode().getPort())
         .errorHandler(log::severe)
         .build()
         .run();
 
     System.clearProperty(LOBBY_HOST);
-    System.clearProperty(LOBBY_PORT);
     System.clearProperty(LOBBY_HTTPS_PORT);
     System.clearProperty(TRIPLEA_NAME);
   }
 
+  private URI lobbyUriFromSystemProps() {
+    return URI.create(
+        HttpLobbyClient.PROTOCOL
+            + System.getProperty(LOBBY_HOST)
+            + ":"
+            + System.getProperty(LOBBY_HTTPS_PORT));
+  }
+
   private static class CouldNotConnectToLobby extends RuntimeException {
     private static final long serialVersionUID = -5946931858867131622L;
 
     CouldNotConnectToLobby() {
       super(
           String.format(
               "Unable to connect to lobby at: %s, port: %s",
-              System.getProperty(LOBBY_HOST), System.getProperty(LOBBY_PORT)));
+              System.getProperty(LOBBY_HOST), System.getProperty(LOBBY_HTTPS_PORT)));
     }
-  }
 
-  synchronized void repostLobbyWatcher() {
-    lobbyWatcher.shutDown();
-    Interruptibles.sleep(3000);
-    createLobbyWatcher();
+    CouldNotConnectToLobby(final FeignException e) {
```

Comment:
It would only be less specific to catch `RuntimeException` or `Exception` or `Throwable`.

`FeignException` is actually not terribly specific, any exceptions thrown by feign client will subclass from that type. 

It's a bad practice to be too overly broad on exception types, right?

Reply:
True, but on exceptions I kinda expact the default `Exception(message, throwable)` constructor 🤷‍♂